Allow paginate(for:) on non-Content Models

stjernegard opened the pull request:
It can be useful to paginate models that aren't necessarily the content themselves
and the code doesn't actually need the Content conformance on this method.

anthonycastelli commented:
I will have a better look at this when I get home next week. There are a couple of things I have to look at for this library too so will hopefully get this all merged in next week.

Sorry it's taken me so long to respond to you on this.

stjernegard committed:
Allow paginate(for:) on non-Content Models

It can be useful to paginate models that aren't necessarily the content themselves
and the code doesn't actually need the Content conformance on this method.

stjernegard commented:
@anthonycastelli, I have rebased my PR
